server/container/util/src/main/java/org/apache/james/util/concurrency/ConcurrentTestRunner.java

```diff
@@ -48,14 +54,19 @@ public ConcurrentRunnableTask(int threadNumber, BiConsumer biConsumer) {
 
         @Override
         public void run() {
+            exception = null;
             countDownLatch.countDown();
             for (int i = 0; i < operationCount; i++) {
                 try {
                     biConsumer.consume(threadNumber, i);
                 } catch (Exception e) {
                     LOGGER.error("Error caught during concurrent testing", e);
+                    exception = e;
```

Comment:
We don't care.... What we care is asserting we did not encounter exceptions.

And in case we did, we rethrow one to allow you to correct your test.

I'm not convinced by your MultipleException....